pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -25,6 +25,14 @@ class TypeHintMatch:
     return_type: list[str] | str | None | object
 
 
+@dataclass
```

Comment:
```suggestion
@dataclass(slots=True)
```
A small performance improvement. Would work for `TypeHintMatch`, too.

Reply:
I have removed this, as we are still using 3.9

Reply:
> I have removed this, as we are still using 3.9

Saw it too after the pylint run failed.